core/src/com/weepingwasp/Main.kt

```diff
@@ -140,6 +154,27 @@ class Main : ApplicationAdapter() {
             }
             PacketID.END_TURN_PLAYER_ATTACKED.id -> {
             }
+            PacketID.END_TURN.id -> {
+                pushEvent(Event(EventType.ENDTURN, hashMapOf("self" to (packet.Content as LinkedTreeMap<*, *>).get("self").toString())))
+            }
+            PacketID.START_TURN.id -> {
+                pushEvent(Event(EventType.STARTTURN, hashMapOf("self" to (packet.Content as LinkedTreeMap<*, *>).get("self").toString())))
+            }
+            PacketID.YOU_DRAW_CARD.id -> {
+                var content = (packet.Content as LinkedTreeMap<*, *>)
+                pushEvent(Event(EventType.DRAWCARD, hashMapOf(
+                    "self" to "true",
+                    "name" to (content.get("card") as LinkedTreeMap<*, *>).get("Name").toString(),
```

Comment:
This is replicated a lot down the function - could you abstract the casting to a new function perhaps?